### PR TITLE
[Help Wanted] [DNM] [Work in Progress] Robust industrial gear

### PR DESCRIPTION
**Pull request by ZeroNetAlpha**
Alrighty. The grate space suit rework is upon us. This is going to be an effort to port over Bay/VORE's RIG suits and space suit taur sprites.

- [x]  Add basic RIG code.
- [ ] Smooth out errors with RIG code, bring it up to CIT/TG standards.
- [ ] Figure out how to get space suits/hardsuits to have taur specific sprites.

### CODE REVIEW

**Comment by ZeroNetAlpha** [comment]
Let me know if you want in on this crazy train.

**Comment by jakeramsay007** [comment]
I've started a port of Bay's rigsuits as well, so I'll be using this for reference and such.

Edit: Nevermind, took one look at Bay's shitcode and said nope to that.

**Comment by ZeroNetAlpha** [comment]
Well, there's always the option of collaborating.

**Comment by deathride58** [comment]
Taur-specific sprites for suits is already handled by #4322 , that PR just needs more sprites for all 6 unique taurtypes. You don't need to stress yourself out for taur altsprites, as I've already done most of the heavy lifting on the code side of things in #4322

**Comment by silicons** [comment]
All I can say is baystation rigsuits are horrible in balance and if you Port this you're going to have to make it balanced to tg/cit combat.

**Comment by Poojawa** [comment]
I have plans for balancing them. 

It's also liable that I'll convert flightsuits to rigs in terms of set up as well. or if just for power code memes

**Comment by silicons** [comment]
flightsuits were made because i wanted to make power armor but i got lazy so i just made flightsuits and called it a day.

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	icon_state = "flash"
+	interface_name = "mounted flash"
+	interface_desc = "Stuns your target by blinding them with a bright light."
+	device_type = /obj/item/device/flash
+
+/obj/item/rig_module/grenade_launcher
+
+	name = "mounted grenade launcher"
+	desc = "A shoulder-mounted micro-explosive dispenser."
+	selectable = 1
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:23]: TRUE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+	var/obj/item/weapon/grenade/new_grenade = new charge.product_type(get_turf(H))
+	H.visible_message("<span class='danger'>[H] launches \a [new_grenade]!</span>")
+	new_grenade.activate(H)
+	new_grenade.throw_at(target,fire_force,fire_distance)
+
+/obj/item/rig_module/mounted
+
+	name = "mounted laser cannon"
+	desc = "A shoulder-mounted battery-powered laser cannon mount."
+	selectable = 1
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:96]: TRUE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+	usable = 1
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:97]: TRUE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+	engage_string = "Configure"
+
+	interface_name = "mounted laser cannon"
+	interface_desc = "A shoulder-mounted cell-powered laser cannon."
+
+	var/gun_type = /obj/item/weapon/gun/energy/lasercannon/mounted
+	var/obj/item/weapon/gun/gun
+
+/obj/item/rig_module/mounted/New()
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:109]: Initialize()

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+	..()
+	gun = new gun_type(src)
+
+/obj/item/rig_module/mounted/engage(atom/target)
+
+	if(!..())
+		return 0
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:116]: FALSE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+	if(!target)
+		gun.attack_self(holder.wearer)
+		return
+
+	gun.Fire(target,holder.wearer)
+	return 1
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:123]: TRUE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+	gun_type = /obj/item/weapon/gun/energy/gun/mounted
+
+/obj/item/rig_module/mounted/taser
+
+	name = "mounted taser"
+	desc = "A palm-mounted nonlethal energy projector."
+	icon_state = "taser"
+
+	usable = 0
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:142]: FALSE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+	desc = "A powerful cutting beam projector."
+	icon_state = "eblade"
+
+	activate_string = "Project Blade"
+	deactivate_string = "Cancel Blade"
+
+	interface_name = "spider fang blade"
+	interface_desc = "A lethal energy projector that can shape a blade projected from the hand of the wearer or launch radioactive darts."
+
+	usable = 0
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:164]: FALSE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+	selectable = 1
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:165]: TRUE

File: modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+	toggleable = 1
```
Comment (silicons) [inline modular_citadel/code/modules/clothing/spacesuits/rig/modules/combat.dm:166]: TRUE

**Review by silicons — CHANGES_REQUESTED** [review]
Instead of stubbornly leaving ~400 comments or whatever on the PR before you're even done, please look at the coding standards I pinned on discord dev channels or whatnot so that we both have less work when you're ready for review.

**Comment by ZeroNetAlpha** [comment]
shit whoops

**Pull request closed without merging**